File: src/main/java/shopping/product/api/dto/ProductDetailResponse.java
package shopping.product.api.dto;

import shopping.product.domain.Product;

public record ProductDetailResponse(Long id, String name, Long price, String imageUrl) {

    public static ProductDetailResponse from(Product product) {
        return new ProductDetailResponse(product.getId(), product.getName(), product.getPrice(), product.getImageUrl());
    }
}

File: src/main/kotlin/shopping/product/application/ProductDetailResponse.kt
package shopping.product.application

import shopping.product.domain.Product
import java.time.LocalDateTime

data class ProductDetailResponse(
    val id: Long,
    val name: String,
    val price: Int,
    val imageUrl: String,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?,
) {
    constructor(product: Product) : this(
        id = product.id,
        name = product.name,
        price = product.price,
        imageUrl = product.imageUrl,
        createdAt = product.createdAt,
        updatedAt = product.updatedAt,
    )
}
